### PR TITLE
Fix build on gcc 11.4.0

### PR DESCRIPTION
**Pull request by zasdfgbnm**

### CODE REVIEW

File: csrc/macros.h [hunk 1]
```diff
@@ -12,6 +12,6 @@
 #include <bits/c++config.h>
 #endif
 
-#if defined(__GLIBCXX__) && __GLIBCXX__ >= 20230000
+#if defined(__GLIBCXX__) && __GLIBCXX__ >= 20230714
```
Comment (zasdfgbnm) [inline csrc/macros.h:15]: I don't know from which version gcc start to support unordered_set with incomplete type. But `20230714` is the one locally on my disk and is tested to work.

**Comment by zasdfgbnm** [comment]
!build

**Review by jjsjann123 — APPROVED** [review]
This looks a little bit scary. https://gcc.gnu.org/bugzilla/show_bug.cgi?id=92770

But stamping since it works in practice.

**Comment by zasdfgbnm** [comment]
> This looks a little bit scary. https://gcc.gnu.org/bugzilla/show_bug.cgi?id=92770

Thanks for the pointer. We can just wait C++ standard to support this feature then we can remove this flag.

**Pull request merged**